crates/requirements-txt/src/lib.rs

```diff
@@ -317,6 +317,27 @@ pub struct RequirementsTxt {
     pub no_index: bool,
 }
 
+/// Calculates column and line based on the cursor and content.
+fn calculate_line_column_pair(content: &str, position: usize) -> (usize, usize) {
+    let mut line = 1;
+    let mut column = 1;
+
+    for (index, char) in content.char_indices() {
```

Comment:
I think we want byte indices rather than char indices here? CC @BurntSushi

Reply:
`char_indices` does actually return byte offsets. The `index` is referring to the index in `content` of where `char` starts. For example, this code:

```rust
let content = "a💩b";
for (i, ch) in content.char_indices() {
    eprintln!("{}:{}", i, ch);
}
```

Has this output:

```
0:a
1:💩
5:b
```

Since `💩` has a UTF-8 encoding that uses 4 bytes.